### DISCUSSION TITLE
How to use the Transform?

### DISCUSSION THREAD

**Discussion by zblesk** [syntax-tree/unist#120] [Q&A] [answered]
I have used the `mdast-util-from-markdown` for some operations and it worked well for me. 

Now I'd like to change the tree before making more operations on it: specifically, I need to inspect and potentially change the `href`s of all links. 

I have noticed the package does contain a Transform type; but I haven't been able to figure out how to do it. 

What is the correct/intended way of doing an operation like this?

**Comment by wooorm** [comment, marked as answer]
plugins contain transforms. Transforms are at the unified level. See the unifiedjs/unified repo for info.

Utilities—the mdast level—do not have a particular interface. You don’t need it. Use `fromMarkdown` to get a tree. Then call `visit` from `unist-util-visit` to walk the tree. That’s it.

**Reply by wooorm** [reply]
if you mean [this transform](https://github.com/syntax-tree/mdast-util-from-markdown/tree/763af0bdca018b9fd89a5e2214d4e792ad1c9b8f#transform), you don‘t need it if you use the package, it’s for extensions.
But I’m also not sure what’s unclear about how it works?

**Reply by zblesk** [reply]
Apparently I've just been trying to use the wrong thing (the Transform class) and hadn't noticed the visitor pattern. 
That explains it. Thanks!
